packages/react-dom/src/server/ReactPartialRenderer.js

```diff
@@ -1200,6 +928,286 @@ class ReactDOMServerRenderer {
     }
   }
 
+  resolve(
+    child: mixed,
+    context: Object,
+  ): {|
+    child: mixed,
+    context: Object,
+  |} {
+    while (React.isValidElement(child)) {
+      // Safe because we just checked it's an element.
+      let element: ReactElement = (child: any);
+      let Component = element.type;
+      if (__DEV__) {
+        pushElementToDebugStack(element);
+      }
+      if (typeof Component !== 'function') {
+        break;
+      }
+      ({child, context} = this.resolveNext(element, context, Component));
+    }
+
+    return {child, context};
```

Comment:
Can we just return the thing returned from `resolveNext` here? Since we already have it.